### PR TITLE
ActiveMerchant: Fix MultiResponse Class

### PR DESCRIPTION
**Pull request by gasb150**
Summary:
---------------------------------------

To be able to pull if there exist the  avs_result and cvv_result properties
in the MultiResponse object, this PR fixes the MultiResponse class by
modifying the logic behind the avs_result and cvv_result method, taking the
last response that has avs_result and cvv_result :code different of nil,
currently, if a transaction returns those values, but the next one returns nil,
the multiresponse methods pulled the AVS and CVV result from the last response transaction.

Local Tests:
---------------------------------------
17 tests, 71 assertions, 1 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

RuboCop:
---------------------------------------
746 files inspected, no offenses detected

### CODE REVIEW

File: test/unit/multi_response_test.rb [hunk 1]
```diff
@@ -172,4 +172,64 @@ def test_handles_ignores_optional_request_result
...
+    m = MultiResponse.run do |r|
+      r.process { r1 }
+      r.process { r2 }
+    end
+    assert_equal [r1, r2], m.responses
+    assert_equal m.avs_result, { 'code' => 'Y', 'message' => 'Street address and 5-digit postal code match.', 'street_match' => 'Y', 'postal_match' => 'Y' }
+    assert_equal m.cvv_result, { 'code' => 'M', 'message' => 'CVV matches' }
+  end
+
+  def test_handles_responses_overwriting_cvv_and_avs_result
```
Comment (Heavyblade) [inline test/unit/multi_response_test.rb:188]: 💬 Comment:
The test title should be "..not_overwriting_existing..." because it is taken the latest cvv/avs if available

File: test/unit/multi_response_test.rb [hunk 2]
```diff
@@ -172,4 +172,64 @@ def test_handles_ignores_optional_request_result
...
+  def test_handles_response_avs_and_cvv_result_with_wrong_values_avs_and_cvv_code
+    r1 = Response.new(true, '1', {}, { avs_result: AVSResult.new(code: '1234567'), cvv_result: CVVResult.new('987654') })
+    r2 = Response.new(true, '2', {})
+    m = MultiResponse.run do |r|
+      r.process { r1 }
+      r.process { r2 }
+    end
+    assert_equal [r1, r2], m.responses
+    assert_equal m.avs_result, { 'code' => '1234567', 'message' => nil, 'street_match' => nil, 'postal_match' => nil }
+    assert_equal m.cvv_result, { 'code' => '987654', 'message' => nil }
```
Comment (Heavyblade) [inline test/unit/multi_response_test.rb:233]: 💬 Comment:

A couple of edge-cases to consider:

- test_getting_cvv_avs_on_multiresponse_with_no_responses

- test_no_overriwrite_cvv_avs_when_use_first_response_flag_enabled

**Review by Heavyblade — COMMENTED** [review]
Looks Pretty Good @gasb150 jut a minor desc correction and a couple of edge cases to consider.

File: test/unit/multi_response_test.rb (hunk 1 above)
Comment (gasb150) [reply test/unit/multi_response_test.rb]: Renamed

File: test/unit/multi_response_test.rb (hunk 2 above)
Comment (gasb150) [reply test/unit/multi_response_test.rb:246]: Greate, I'll add them.

**Review by mbreenlyles — COMMENTED** [review]
This looks good to me! great test coverage too! The only thing I noticed were some rubocop failures, the one I can see is:

`test/unit/multi_response_test.rb:215:1: C: Layout/TrailingWhitespace: Trailing whitespace detected. (https://github.com/rubocop-hq/ruby-style-guide#no-trailing-whitespace)`

I think you just need to clear that whitespace on line 215 and it should be good

**Review by mbreenlyles — APPROVED** [review]
looks great! tagging @curiousepic for the last approval 🤠

**Review by curiousepic — APPROVED** [review]
Nice, LGTM, thanks for the thorough testcases! I also ran some remote gateway tests that include multiple avs/cvv_result assertions like Authorize.net and Braintree, which were good. Will merge on Monday.

**Comment by curiousepic** [comment]
Sorry for lagging on this - will try to get it in tomorrow.
